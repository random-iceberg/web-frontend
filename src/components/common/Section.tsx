import React from "react";

/**
 * Props for the Section component. Extends standard HTML div attributes.
 * @typedef {object} SectionProps
 * @property {React.ReactNode} children - Content to be rendered inside the section.
 * @property {string} [className] - Additional CSS classes to apply to the section container.
 */
type SectionProps = React.HTMLAttributes<HTMLDivElement> & {
  children?: React.ReactNode;
  className?: string;
};

/**
 * A simple wrapper for content sections with consistent padding and max-width.
 * Uses a standard HTML <section> tag.
 * Accepts standard HTML attributes like `id`.
 *
 * @component
 * @param {SectionProps} props - Props for the Section component.
 * @example
 * <Section id="features" className="bg-gray-100">
 *   <h2>Features</h2>
 *   <p>Feature details go here.</p>
 * </Section>
 */
const Section: React.FC<SectionProps> = ({
  children,
  className = "",
  ...rest // Pass down other div attributes like id
}) => (
  <section
    className={`container mx-auto px-4 py-16 ${className}`.trim()} // Default py-16 padding
    {...rest}
  >
    {children}
  </section>
);

export default Section;
