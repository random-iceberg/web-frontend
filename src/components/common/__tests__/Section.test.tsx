import React from 'react';
import { render } from '@testing-library/react';
import Section from '../Section';

describe('Section Component', () => {
  it('renders correctly with children', () => {
    const { asFragment } = render(
      <Section>
        <h1>Section Title</h1>
        <p>This is section content.</p>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with children and className', () => {
    const { asFragment } = render(
      <Section className="my-custom-section bg-blue-100" id="test-section">
        <h2>Another Section</h2>
      </Section>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no children', () => {
    // A section can technically be empty, though usually it will have content.
    const { asFragment } = render(<Section id="empty-section" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
