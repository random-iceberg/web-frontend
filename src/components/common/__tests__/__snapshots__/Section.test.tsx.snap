// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component renders correctly with children 1`] = `
<DocumentFragment>
  <section
    class="container mx-auto px-4 py-16"
  >
    <h1>
      Section Title
    </h1>
    <p>
      This is section content.
    </p>
  </section>
</DocumentFragment>
`;

exports[`Section Component renders correctly with children and className 1`] = `
<DocumentFragment>
  <section
    class="container mx-auto px-4 py-16 my-custom-section bg-blue-100"
    id="test-section"
  >
    <h2>
      Another Section
    </h2>
  </section>
</DocumentFragment>
`;

exports[`Section Component renders correctly with no children 1`] = `
<DocumentFragment>
  <section
    class="container mx-auto px-4 py-16"
    id="empty-section"
  />
</DocumentFragment>
`;
